zeebe/exporters/operate-exporter/src/main/java/io/camunda/zeebe/operate/exporter/ElasticsearchConnector.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.http.ssl.TrustStrategy;
+import org.elasticsearch.client.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO this is a copy-paste from io.camunda.operate.connect.ElasticsearchConnector
+// when we start to use Spring in exporter we should refactor to reuse it
+public class ElasticsearchConnector {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ElasticsearchConnector.class);
+
+  private ElasticsearchClient elasticsearchClient;
+
+  public ElasticsearchClient elasticsearchClient(final ElasticsearchProperties elsConfig) {
+    LOGGER.debug("Creating ElasticsearchClient ...");
+    final RestClientBuilder restClientBuilder = RestClient.builder(getHttpHost(elsConfig));
+    if (elsConfig.getConnectTimeout() != null || elsConfig.getSocketTimeout() != null) {
+      restClientBuilder.setRequestConfigCallback(
+          configCallback -> setTimeouts(configCallback, elsConfig));
+    }
+    final RestClient restClient =
+        restClientBuilder
+            .setHttpClientConfigCallback(
+                httpClientBuilder -> configureHttpClient(httpClientBuilder, elsConfig))
+            .build();
+
+    // Create the transport with a Jackson mapper
+    final ElasticsearchTransport transport =
+        new RestClientTransport(restClient, new JacksonJsonpMapper());
+
+    // And create the API client
+    elasticsearchClient = new ElasticsearchClient(transport);
+    if (!checkHealth(elasticsearchClient, elsConfig)) {
+      LOGGER.warn("Elasticsearch cluster is not accessible");
+    } else {
+      LOGGER.debug("Elasticsearch connection was successfully created.");
+    }
+    return elasticsearchClient;
+  }
+
+  public boolean checkHealth(
+      final ElasticsearchClient elasticsearchClient, final ElasticsearchProperties elsConfig) {
+    try {
+      return RetryOperation.<Boolean>newBuilder()
+          .noOfRetry(50)
+          .retryOn(
+              IOException.class,
+              co.elastic.clients.elasticsearch._types.ElasticsearchException.class)
+          .delayInterval(3, TimeUnit.SECONDS)
+          .message(
+              String.format(
+                  "Connect to Elasticsearch cluster [%s] at %s",
+                  elsConfig.getClusterName(), elsConfig.getUrl()))
+          .retryConsumer(
+              () -> {
+                final HealthResponse healthResponse = elasticsearchClient.cluster().health();
+                LOGGER.info("Elasticsearch cluster health: {}", healthResponse.status());
+                return healthResponse.clusterName().equals(elsConfig.getClusterName());
+              })
+          .build()
+          .retry();
+    } catch (final Exception e) {
+      throw new OperateRuntimeException("Couldn't connect to Elasticsearch. Abort.", e);
+    }
+  }
+
+  //  @PreDestroy
```

Comment:
If we don't need this commented-out code I think we should remove it

Reply:
We will need it when converting to Spring, but I will remove for now 👍

Reply:
If we want to keep the function for now to re-use when spring is added that's fine, let's just add a comment that explains that